🌱 Add logging guidelines (#6969)
kubernetes-sigs/cluster-api: REVIEWING.md

### Logging

> While developing controllers in Cluster API it is required add logging so the system can be observed
> and all the required info is available when troubleshooting issues.

Reviewer: ```suggestion
While developing controllers in Cluster API a key requirement is to add logging to observe the system and 
to help troubleshooting issues.
```

> and all the required info are available when troubleshooting issues.

Reviewer 2: ```suggestion
and all the required info is available when troubleshooting issues.
```

> - For CAPI controllers see TODO.

Reviewer: ?

Author: see https://github.com/kubernetes-sigs/cluster-api/pull/6969#discussion_r928828230

> - For CAPI controllers see TODO.

Reviewer 2: Can we just force merge with the correct link? This is pretty bad UX for adding pages to the book!

> - For CAPI controllers see TODO.

Author: REVIEWING.md is an edge case, because this page is defined outside of the book but also imported in the book.
So, in order to make this page to work from outside the book you need to use the generated url of the target page https://cluster-api.sigs.k8s.io/developer/logging.html, that will exists only when the book is merged.
